build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RunAndroidEmulatorCheckBootTimes.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Android.BuildTools.PrepTasks;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class RunAndroidEmulatorCheckBootTimes : Task
+	{
+		public string CheckBootTimesPath { get; set; }
+		public string DeviceName { get; set; } = "XamarinAndroidTestRunner";
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.High, $"Task {nameof (RunAndroidEmulatorCheckBootTimes)}");
+			Log.LogMessage (MessageImportance.High, $"  {nameof (CheckBootTimesPath)}: {CheckBootTimesPath}");
+			Log.LogMessage (MessageImportance.High, $"  {nameof (DeviceName)}: {DeviceName}");
+
+			var fileInfo = new FileInfo (CheckBootTimesPath);
+			Log.LogMessage (MessageImportance.High, $"CheckBootTimesPath:  {fileInfo.FullName}");
+
+			if (!fileInfo.Exists) {
+				Log.LogError ($"Unable to find {fileInfo.FullName}.");
+				return !Log.HasLoggedErrors;
+			}
+
+			bool finishAsExpected = false;
+			var success = RunProcess (
+				fileInfo.FullName,
+				$"-DeviceName {DeviceName}",
+				5 * 60000,
+				(string data, ManualResetEvent mre) => {
+					Log.LogMessage (MessageImportance.High, $"CheckBootTimes ({DateTime.UtcNow}): {data}");
+					if (!string.IsNullOrWhiteSpace (data) && data.IndexOf ("Check-Boot-Times Done.", StringComparison.OrdinalIgnoreCase) != -1) {
+						finishAsExpected = true;
+						mre.Set ();
+					}
+
+					return true;
+				});
+
+			if (!success || !finishAsExpected) {
+				Log.LogError ("check-boot-times did not run as expected, please see logs for more info.");
+				CloseProcesses ();
+				return false;
+			}
+
+			Log.LogMessage (MessageImportance.High, $"Done with RunAndroidEmulatorCheckBootTimes task.");
+			return true;
+		}
+
+
+		static void CloseProcesses ()
+		{
+			// Try to close emulator
+			for (int i = 0; i < 10; i++) {
+				try {
+					var proc = System.Diagnostics.Process.GetProcesses ().FirstOrDefault (p => p.ProcessName.IndexOf ("qemu") != -1 || p.ProcessName.IndexOf ("emulator") != -1);
```

Comment:
This does a culture-sensitive search.  This should probably use the [`string.IndexOf(string, StringComparison)` overload](https://docs.microsoft.com/en-us/dotnet/api/system.string.indexof?view=netframework-4.8#System_String_IndexOf_System_String_System_StringComparison_).  We might also want `StringComparison.OrdinalIgnoreCase`, Just In Case on Windows...

Reply:
Given that this LINQ expression is used twice in this method, it should probably be split up into a separate `GetAndroidEmulator()` method.

Reply:
good catch! Fixed!